controllers/scopeinstance_controller.go

```diff
@@ -103,208 +134,197 @@ func (r *ScopeInstanceReconciler) Reconcile(ctx context.Context, req ctrl.Reques
-							scopeInstanceHashKey:          util.HashObject(in.Spec),
-							scopeTemplateHashKey:          util.HashObject(st.Spec),
-							clusterRoleBindingGenerateKey: cr.GenerateName,
-						},
-						OwnerReferences: []metav1.OwnerReference{{
-							APIVersion: in.APIVersion,
-							Kind:       in.Kind,
-							Name:       in.GetObjectMeta().GetName(),
-							UID:        in.GetObjectMeta().GetUID(),
-						}},
-					},
-					Subjects: cr.Subjects,
-					RoleRef: rbacv1.RoleRef{
-						Kind:     "ClusterRole",
-						Name:     cr.GenerateName,
-						APIGroup: "rbac.authorization.k8s.io",
-					},
-				}
+		return nil
+	}
 
-				rbList := &rbacv1.RoleBindingList{}
-				if err := r.Client.List(ctx, rbList, &client.ListOptions{
-					Namespace: namespace,
-				}, client.MatchingLabels{
-					scopeInstanceUIDKey:           string(in.GetUID()),
-					scopeTemplateUIDKey:           string(st.GetUID()),
-					clusterRoleBindingGenerateKey: cr.GenerateName,
-				}); err != nil {
-					return err
-				}
+	log.Log.Info("Updating existing rb", "namespaced", rbList.Items[0].GetNamespace(), "name", rbList.Items[0].GetName())
 
-				if len(rbList.Items) > 1 {
-					return fmt.Errorf("more than one roleBinding found for ClusterRole %s", cr.GenerateName)
-				}
+	existingRB := &rbList.Items[0]
 
-				// GenerateName is immutable, so create the object if it has changed
-				if len(rbList.Items) == 0 {
-					if err := r.Client.Create(ctx, rb); err != nil {
-						return err
-					}
-					continue
-				}
+	if util.IsOwnedByLabel(existingRB.DeepCopy(), in) &&
+		reflect.DeepEqual(existingRB.Subjects, rb.Subjects) &&
+		reflect.DeepEqual(existingRB.Labels, rb.Labels) {
+		log.Log.Info("existing RoleBinding does not need to be updated")
+		return nil
+	}
 
-				log.Log.Info("Updating existing rb", "namespaced", rbList.Items[0].GetNamespace(), "name", rbList.Items[0].GetName())
+	u, err := r.patchConfigForRoleBinding(existingRB, rb)
+	if err != nil {
+		return err
+	}
+
+	// server-side apply patch
+	if err := r.patchBinding(ctx, u); err != nil {
+		return err
+	}
 
-				existingRB := &rbList.Items[0]
+	return nil
+}
 
-				if util.IsOwnedByLabel(existingRB.DeepCopy(), in) &&
-					reflect.DeepEqual(existingRB.Subjects, rb.Subjects) &&
-					reflect.DeepEqual(existingRB.Labels, rb.Labels) {
-					log.Log.Info("existing ClusterRoleBinding does not need to be updated")
-					return nil
-				}
-				existingRB.Labels = rb.Labels
-				existingRB.OwnerReferences = rb.OwnerReferences
-				existingRB.Subjects = rb.Subjects
-
-				// server-side apply patch
-				existingRB.ManagedFields = nil
-				if err := r.Client.Patch(ctx,
-					existingRB,
-					client.Apply,
-					client.FieldOwner(siCtrlFieldOwner),
-					client.ForceOwnership); err != nil {
-					return err
-				}
-			}
+func (r *ScopeInstanceReconciler) patchConfigForRoleBinding(oldRb *rbacv1.RoleBinding, rb *rbacv1.RoleBinding) (*unstructured.Unstructured, error) {
```

Comment:
The idea here is that we aren't using concrete API objects, and rather using applyconfigurations that are being converted to `unstructured.Unstructured` objects to be used for submitting changes via a server-side apply patch request.

Now that I think about it, it may be easier to just do this with an `unstructured.Unstructured` object than trying to do all this weird applyconfiguration stuff.

Reply:
Updated in commit https://github.com/operator-framework/oria-operator/pull/48/commits/f80fcf25a2b332882d4f9f4d75a37ac1f40b6d4f